editor/plugins/tiles/tile_set_atlas_source_editor.cpp

```diff
@@ -2259,54 +2263,61 @@ void TileSetAtlasSourceEditor::edit(Ref<TileSet> p_tile_set, TileSetAtlasSource
 	_update_current_tile_data_editor();
 }
 
-void TileSetAtlasSourceEditor::init_source() {
+void TileSetAtlasSourceEditor::init_source(Vector<Ref<TileSetAtlasSource>> atlases) {
 	tool_setup_atlas_source_button->set_pressed(true);
+	atlases_to_auto_create_tiles = atlases;
 	confirm_auto_create_tiles->popup_centered();
 }
 
 void TileSetAtlasSourceEditor::_auto_create_tiles() {
-	if (!tile_set_atlas_source) {
-		return;
-	}
+	for (Ref<TileSetAtlasSource> atlas_source : atlases_to_auto_create_tiles) {
+		if (atlas_source.is_valid()) {
+			Ref<Texture2D> texture = atlas_source->get_texture();
+			if (texture.is_valid()) {
+				Vector2i margins = atlas_source->get_margins();
+				Vector2i separation = atlas_source->get_separation();
+				Vector2i texture_region_size = atlas_source->get_texture_region_size();
+				Size2i grid_size = atlas_source->get_atlas_grid_size();
+				EditorUndoRedoManager *undo_redo = EditorUndoRedoManager::get_singleton();
+				undo_redo->create_action(TTR("Create tiles in non-transparent texture regions"));
+				for (int y = 0; y < grid_size.y; y++) {
+					for (int x = 0; x < grid_size.x; x++) {
+						// Check if we have a tile at the coord
```

Comment:
```suggestion
						// Check if we have a tile at the coord.
```
Code style requires that comments start with a capital letter and ends in a period

Reply:
Updated